### PR TITLE
Handle autoscoped constructor args with no UUID

### LINKED ISSUE #11325: Unexpected Type_Error/Method unknown on project startup
Encountering regularly in a project a missing method Panic such as:
Screenshot from 2024-10-15 09-56-59

Surprisingly, when we refresh the project, it disappears. Similarly, if (IR) caches for the project are there, the project will also continue to load without hiccups.

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

An application involving >1 autoscoped atom constructor arguments with no ID would lead to a silent type error in GUI. It was silent because once IdMap gets updated, the original type error disappears and users are left with a No_Such_Method on a Panic.

The type error may occur because the compiler was inferring the same UUID for autoscoped constructors. Args with UUID are cached therefore a type confict might occur on the second (or later) argument.

Added a unit test case demonstrating the problem (previously it would fail). The search path is now a bit more careful when inferring UUIDs for arguments.

Closes #11325.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

**Comment by hubertp** [comment]
What probably remains is to make sure that once IdMap arrives, expression updates are being sent. The second half of the unit test demonstrates that this is not necessarily the case.

**Review by JaroslavTulach — APPROVED** [review]

File: engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala [hunk 1]
```diff
@@ -4074,6 +4074,238 @@ class RuntimeVisualizationsTest extends AnyFlatSpec with Matchers {
...
+          |    s = Singleton.S 1
+          |    x = s.test ..A ..B
+          |    x
+          |""".stripMargin.linesIterator.mkString("\n")
+      val contents = metadata.appendToCode(code)
+      val mainFile = context.writeMain(contents)
+
+      metadata.assertInCode(idS, code, "Singleton.S 1")
+      metadata.assertInCode(idX, code, "s.test ..A ..B")
+      metadata.assertInCode(idRes, code, "x")
```
Comment (JaroslavTulach) [inline engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:4139]: +1

**Comment by hubertp** [comment]
Re my previous comment regarding lack of expression updates: https://github.com/enso-org/enso/pull/11354#issuecomment-2420568916

I imagine the issue is that once `UnresolvedConstructor` builds the application for autoscoped constructors we are hitting the cached version in https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/UnresolvedConstructor.java#L213 and no further construction happens. So I would essentially have to invalidate that when IdMap contains new UUID for the given argument.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/UnresolvedConstructor.java">enso/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/UnresolvedConstructor.java at develop · enso-org/enso</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by hubertp** [comment]
> Re my previous comment regarding lack of expression updates: [#11354 (comment)](https://github.com/enso-org/enso/pull/11354#issuecomment-2420568916)

I was wrong. It's a feature.

**Pull request merged**